### PR TITLE
Sync persistance_improvement branch with master

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #<Issue Number>

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/variable_number_of_resource_paths.bal [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    isolated resource function get def/[int id]/[string path]/[string p2] (string s) returns string = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.Environment", "io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString"]
+    } external;
+
+    isolated resource function get ghi/[int id]/[string p1]/[string p2]/[int ...ids] (string s) returns string = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.Environment", "io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString"]
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/variable_number_of_resource_paths.bal:65]: Shall we add a test without param types and with only one function with given name.

File: tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/variable_number_of_resource_paths.bal [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import ballerina/jballerina.java;
+import ballerina/test;
+
+public client isolated class Client {
+
+    public isolated function init() {
+
+    }
+
+    isolated resource function get abc/[string ...path] () returns int = @java:Method {
```
Comment (Nadeeshan96) [inline tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/variable_number_of_resource_paths.bal:26]:
```suggestion
    isolated resource function get abc/[string ...path]() returns int = @java:Method {
```

Comment (Nadeeshan96) [reply tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/variable_number_of_resource_paths.bal:26]: I think we should not keep a space here as seen in https://ballerina.io/spec/http/#233-path-parameter. Shall we change in all the places?

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Pull request merged**